tests/scripts/thread-cert/Cert_5_2_05_AddressQuery.py

```diff
@@ -30,110 +30,130 @@
+
+        self.sniffer = config.create_default_thread_sniffer(SNIFFER)
+        self.sniffer.start()
 
     def tearDown(self):
+        self.sniffer.stop()
+        del self.sniffer
+
         for node in list(self.nodes.values()):
             node.stop()
         del self.nodes
 
     def test(self):
+        # 1. LEADER: Configure the Leader to be a DHCPv6  Border Router for prefix 2001::.
         self.nodes[LEADER].start()
         self.nodes[LEADER].set_state('leader')
         self.assertEqual(self.nodes[LEADER].get_state(), 'leader')
+        self.nodes[LEADER].add_prefix('2001::/64', 'pdros')
+        self.nodes[LEADER].register_netdata()
 
-        self.nodes[BR1].start()
+        # 2. BR: Attach the Border_Router to the network and configure On-Mesh Prefix.
+        self.nodes[BR].start()
         time.sleep(5)
-        self.assertEqual(self.nodes[BR1].get_state(), 'router')
-
-        self.nodes[BR1].add_prefix('2001:2:0:3::/64', 'paros')
-        self.nodes[BR1].add_prefix('2001:2:0:4::/64', 'paros')
-        self.nodes[BR1].register_netdata()
+        self.assertEqual(self.nodes[BR].get_state(), 'router')
+        self.nodes[BR].add_prefix('2002::/64', 'paros')
+        self.nodes[BR].register_netdata()
+
+        # 3. All: Ensure topology is formed correctly without the DUT_REED.
+        for i in range(2, 17):
+            if i == BR:
+                continue
+            self.nodes[i].start()
+            time.sleep(5)
+            self.assertEqual(self.nodes[i].get_state(), 'router')
 
         self.nodes[ED1].start()
         time.sleep(5)
         self.assertEqual(self.nodes[ED1].get_state(), 'child')
 
-        self.nodes[REED].start()
+        # 4. DUT_REED: Cause the DUT_REED to attach to ROUTER1.
+        self.nodes[DUT_REED].start()
         time.sleep(5)
-        self.assertEqual(self.nodes[REED].get_state(), 'child')
+        time.sleep(ROUTER_SELECTION_JITTER)
 
-        self.nodes[ROUTER2].start()
-        time.sleep(5)
-        self.assertEqual(self.nodes[ROUTER2].get_state(), 'router')
+        reed_messages = self.sniffer.get_messages_sent_by(DUT_REED)
 
-        self.nodes[ED2].start()
-        time.sleep(5)
-        self.assertEqual(self.nodes[ED2].get_state(), 'child')
+        # The DUT_REED must not send a coap message here.
+        msg = reed_messages.does_not_contain_coap_message()
+        assert msg is True, "Error: The REED sent an Address Solicit Request"
 
-        self.nodes[ED3].start()
-        time.sleep(5)
-        self.assertEqual(self.nodes[ED3].get_state(), 'child')
+        # 5. Enables a link between the DUT and BR to create a one-way link.
+        self.nodes[DUT_REED].add_whitelist(self.nodes[BR].get_addr64())
+        self.nodes[BR].add_whitelist(self.nodes[DUT_REED].get_addr64())
+
+        # 6. Send an ICMPv6 Echo Request from ED_1 to REED_1 using ML-EID/2001:: EID/2002:: EID to test if the DUT_REED would send a properly formatted Address Notification or not.
```

Comment:
Suggestion:
`# 6. Verify DUT_REED would send Address Notification when ping to its ML-EID.`

Reply:
fixed.